warehouse/ingest-core/src/main/java/datawave/ingest/data/config/ingest/WhindexFieldIngestHelper.java

```diff
@@ -0,0 +1,138 @@
+
+        Map<String,String> properties = config.getPropsWithPrefix(commonPrefix);
+        Map<String,WhindexConfig> groupingsToConfigs = new HashMap<>();
+
+        for (Map.Entry<String,String> entry : properties.entrySet()) {
+            String[] parts = entry.getKey().split("\\.");
+            String groupID = parts[0];
+            String property = parts[1];
+
+            WhindexConfig whindexConfig = groupingsToConfigs.computeIfAbsent(groupID, (k) -> new WhindexConfig());
+            switch (property) {
+                case VALUE_FIELD:
+                    whindexConfig.setValueField(entry.getValue());
+                    break;
+                case SRC_FIELD:
+                    whindexConfig.setSourceField(entry.getValue());
+                    break;
+                case DELETE_SRC_FIELD:
+                    whindexConfig.setOverloaded(Boolean.parseBoolean(entry.getValue()));
+                    if (whindexConfig.isOverloaded()) {
+                        overloadedFields.add(whindexConfig.getSourceField());
+                    }
+                    break;
+                case DST_FIELD:
+                    whindexConfig.setDestField(entry.getValue());
+                    break;
+                case VALUES:
+                    whindexConfig.setValues(List.of(entry.getValue().split(",")));
+                    break;
+                default:
+                    String originalProperty = commonPrefix + groupID + "." + property;
+                    log.warn("Unexpected whindex property given:" + originalProperty + "=" + entry.getValue());
+            }
+        }
+
+        // add to fieldDefinitions and VFToWC
+        groupingsToConfigs.values().forEach((wc) -> {
+            valueFieldsToWhindexConfigs.put(wc.getValueField(), wc);
+        });
+
+    }
+
+    @Override
+    public Multimap<String,NormalizedContentInterface> getWhindexFields(Multimap<String,NormalizedContentInterface> eventMap) {
+
+        Multimap<String,NormalizedContentInterface> whindicesInEventMap = HashMultimap.create();
+
+        // Get all the wc that have both vf and sf in the eventMap
+        List<WhindexConfig> matchingConfigs = valueFieldsToWhindexConfigs.entries().stream()
+                        .filter(entry -> eventMap.containsKey(entry.getValue().getValueField()) && eventMap.containsKey(entry.getValue().getSourceField()))
+                        .map(Map.Entry::getValue).collect(Collectors.toList());
+
+        // Check that the eventMap entry has EITHER an EventField or IndexedField in common with the wc's VALUES
+        for (WhindexConfig curWhindexConfig : matchingConfigs) {
+            Collection<NormalizedContentInterface> relatedValueEventContents = eventMap.get(curWhindexConfig.getValueField());
+            // if any of the NCI's have either EF or IF that's GOOD!
+            boolean containsAnyMatchingValue = relatedValueEventContents.stream()
```

Comment:
You can create a local variable that contains the result of `curWhindexConfig.getValues()`, and then reference that when checking if the set contains either the event field value or index field value. This will avoid multiple calls to `curWhindexConfig.getValues().`

Additionally, please use `curr` instead of `cur` when using as an abbreviation for current.